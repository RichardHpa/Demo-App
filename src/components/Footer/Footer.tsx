function Copyright() {
  return (
    <p>
      {'Copyright Â© '}
      {new Date().getFullYear()}
    </p>
  )
}

export const Footer = () => {
  return (
    <footer className='px-2 sm:px-4 mt-auto'>
      <div className='container mx-auto border-t py-4'>
        <Copyright />
      </div>
    </footer>
  )
}
