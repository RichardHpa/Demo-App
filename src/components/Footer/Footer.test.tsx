import { render, screen } from '@testing-library/react'

import { Footer } from './Footer'

jest.useFakeTimers().setSystemTime(new Date('2020-01-01'))

describe('Footer', () => {
  test('renders footer', () => {
    render(<Footer />)
    expect(screen.getByText('Copyright Â© 2020')).toBeInTheDocument()
  })

  test('renders footer snapshot', () => {
    const { container } = render(<Footer />)
    expect(container).toMatchSnapshot()
  })
})
