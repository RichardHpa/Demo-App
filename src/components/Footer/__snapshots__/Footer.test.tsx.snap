// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders Footer snapshot 1`] = `
<div>
  <footer
    class="px-2 sm:px-4 mt-auto"
  >
    <div
      class="container mx-auto border-t py-4"
    >
      <p>
        Copyright Â© 
        2020
      </p>
    </div>
  </footer>
</div>
`;
